### DISCUSSION TITLE
slow simulation when recombination rate is high

### DISCUSSION THREAD

**Discussion by Devashish13** [tskit-dev/msprime#2103] [General]
Hi,

I am simulating whole genome data for human population. the parameters i used are as follows

number of diploids = 200
mutation rate = 1.25e-8
recombination rate = 8e-7 (estimated using smc++)
sequence length = chromosome length
demographic model = neutral one population model, specified using effecting population size changes obtained through smc++.

for chromosome 22,
recombination rate = 1e-8, time taken = 12 seconds
recombination rate = 1e-7, time taken = 12 minutes
recombination rate = 8e-7 still going on it's been 33 hours as of raising this issue.

High recombination rate will slow the simulation but i have no idea how much time it gonna take.

kindly let me know what is going on here!!
is there any way to make it faster?

thanks,
Devashish

**Comment by grahamgower** [comment]
Hi @Devashish13,
Yes, this is a known issue (#1651). See [Baumdicker et al. 2022](https://academic.oup.com/genetics/article/220/3/iyab229/6460344) (the msprime-1.0 paper), in particular the "Recombination" section and the "Time complexity of Hudson’s algorithm" section of the Appendix. The latter provides a graph that you should be able to use to estimate the running time.

**Reply by Devashish13** [reply]
Thanks @grahamgower 

I went through that. I am simulating the data under a demographic model having population expansion in the past(for ~3000 generations) followed by constant population size. 

I have simulated the data under constant population size(r = 8e-7) and expected time can be estimated using the graph provided in paper and its relatively fast. But with such demographic model I am using, the scaled time is very large(~62 hours, 21st chromosome)

The CPU I have used is Intel(R) Xeon(R) CPU E5-2683 v3 @ 2.00GHz

is there any suggestion on reducing the simulation time in some way?

Thanks,
Devashish

**Reply by grahamgower** [reply]
Maybe you could try using the DTWF model for the recent past, and switching to the hudson model at older times?
https://tskit.dev/msprime/docs/stable/ancestry.html#sec-ancestry-models-dtwf
Nelson et al. 2020, https://doi.org/10.1371/journal.pgen.1008619

**Reply by Devashish13** [reply]
Hi @grahamgower , hope you are doing great!!

Instead of specifying the population recombination rate estimated through SMC++. I have used the 1000 genome recombination map to simulate the whole genome data. The algorithm worked well as the simulated data was close to expected real data and time efficient as well.
